src/main/java/roboy/context/PeriodicUpdater.java

```diff
@@ -10,22 +10,20 @@
  * An implementation of the UpdatePolicy which performs regular updates on a target object.
  * The method update() needs to be implemented in the subclass.
  *
- * @param <T> The class of the target object.
+ * @param <Target> The class of the target object.
  */
-public abstract class IntervalUpdater<T> extends ExternalUpdater {
-    protected final T target;
+public abstract class PeriodicUpdater<Target> extends ExternalUpdater {
+    protected final Target target;
     protected final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
fancy stuff here :)
``scheduler`` & ``updateFrequencySeconds`` could be private unless you plan to access them from somewhere else.

Reply:
In some cases, the variables are accessed by subclasses, tests, etc.

1) updateFrequencySeconds - we can overwrite the standard updating interval in subclass depending the individual case (and also made it public static for testing purposes).

2) scheduler - not sure why I made it protected, will change to private.

3) target - you figured that one out already ;) But just for completeness, this will be needed when implementing the custom update() method in a subclass.